experimenter/tests/integration/nimbus/pages/base.py

```diff
@@ -19,6 +22,11 @@ def _wait_for_it(selenium):
                 self.wait_for_page_to_load()
                 selenium.find_element(*locator)
             except NoSuchElementException:
+                sleep_thread = threading.Thread(
+                    target=self.non_blocking_sleep, args=(10,)
+                )
+                sleep_thread.start()
+                sleep_thread.join()  # Wait for the thread to finish sleeping
```

Comment:
What's this for?

Reply:
that's the little improvement I made to help the test run locally successfully, what I was noticing that refresh was always going on the main thread and was not letting RS to do task, as soon as refreshes stopped, RS kicked in but at that time test was getting failed already because they were not able to find the right status, so with this optimization tests works 💯